File: examples/fake_git.py
from argparse import ArgumentParser

if __name__ == "__main__":
    parser = ArgumentParser()
    subparsers = parser.add_subparsers(required=True)
    clone = subparsers.add_parser("clone")
    clone.add_argument("repo")
    commit = subparsers.add_parser("commit")
    commit.add_argument("-m", type=str, required=True)
    parser.parse_args()

File: examples/fake_git.rs
use std::path::PathBuf;

use clap::{Parser, Subcommand};

#[derive(Parser, Debug)]
struct Cli {
    #[command(subcommand)]
    git_subcommand: GitSubcommand,
}

#[derive(Subcommand, Debug)]
enum GitSubcommand {
    Clone { repo: String },
    Commit(GitCommitArgs),
}

#[derive(Parser, Debug)]
struct GitCommitArgs {
    #[arg(short)]
    message: String,
}

fn main() {
    let cli = Cli::parse();
    dbg!(cli);
}
